### PR TITLE
stream sync response

### PR DESCRIPTION
**Pull request by sitaram-kalluri**

### CODE REVIEW

File: at_lookup/lib/src/at_lookup_impl.dart [hunk 1]
```diff
@@ -36,7 +36,7 @@ class AtLookupImpl implements AtLookUp {
 
   var privateKey;
 
-  var _cramSecret;
+  var cramSecret;
```
Comment (murali-shris) [inline at_lookup/lib/src/at_lookup_impl.dart:39]: why has this var been made public ?

File: at_lookup/lib/src/connection/outbound_sync_message_listener.dart [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  SyncMessageListener(connection) : super(connection);
+
+  @override
+  Future<void> messageHandler(sync_data) async {
+    // If sync_data contains only '@', return.
+    if (sync_data.length == 1 && sync_data.first == atConst) {
+      return;
+    }
+    // if sync_data last is '@' and contains a new line.
+    if (sync_data.last == atConst && sync_data.contains(10)) {
```
Comment (murali-shris) [inline at_lookup/lib/src/connection/outbound_sync_message_listener.dart]: replace 10 with a constant

File: at_lookup/lib/src/connection/outbound_sync_message_listener.dart [hunk 3]
```diff
@@ -0,0 +1,54 @@
+import 'dart:collection';
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:at_lookup/src/connection/outbound_message_listener.dart';
+
+class SyncMessageListener extends OutboundMessageListener {
+  Queue pendingData = Queue<List<int>>();
+  var atConst = '@'.codeUnitAt(0);
```
Comment (murali-shris) [inline at_lookup/lib/src/connection/outbound_sync_message_listener.dart]: this is const..not a var

File: at_lookup/lib/src/connection/outbound_sync_message_listener.dart [hunk 4]
```diff
@@ -0,0 +1,54 @@
+import 'dart:collection';
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:at_lookup/src/connection/outbound_message_listener.dart';
+
+class SyncMessageListener extends OutboundMessageListener {
```
Comment (murali-shris) [inline at_lookup/lib/src/connection/outbound_sync_message_listener.dart:7]: you can add unit test for this class..check pendingData, verify whether sync callback is called

File: at_lookup/lib/src/at_lookup_impl.dart (hunk 1 above)
Comment (sitaram-kalluri) [reply at_lookup/lib/src/at_lookup_impl.dart:39]: To make it visible in the AtLookupSync class (Child class of this class).

File: at_lookup/lib/src/connection/outbound_sync_message_listener.dart (hunk 2 above)
Comment (sitaram-kalluri) [reply at_lookup/lib/src/connection/outbound_sync_message_listener.dart]: Replaced with a final variable. Getting compile issue when having it as constant.

File: at_lookup/lib/src/connection/outbound_sync_message_listener.dart (hunk 3 above)
Comment (sitaram-kalluri) [reply at_lookup/lib/src/connection/outbound_sync_message_listener.dart]: Replaced with a final variable. Getting compile issue when having it as constant.

File: at_lookup/lib/src/connection/outbound_sync_message_listener.dart (hunk 4 above)
Comment (cpswan) [reply at_lookup/lib/src/connection/outbound_sync_message_listener.dart:7]: Where are the tests?

**Review by cpswan — COMMENTED** [review]
I can't see where tests are being run by the GitHub Actions?

**Comment by kalluriramkumar** [comment]
> I can't see where tests are being run by the GitHub Actions?

Following is the link to changes in github action's to unit test's in at_lookup : https://github.com/atsign-foundation/at_libraries/pull/51/commits/2e13eba477e9e5b79b7454891221136211e5f282. Attaching the screenshot of github actions's as well
image

**Review by cpswan — APPROVED** [review]

**Review by murali-shris — APPROVED** [review]

**Pull request merged**